### PR TITLE
itemgroup: turn OwoItemSettings into an interface injection

### PR DESCRIPTION
**Pull request by BasiqueEvangelist**
Since this breaks source and binary compatibility, this should probably be scheduled for a major release

### CODE REVIEW

**Comment by Jab125** [comment]
Might be worth namespacing the injected methods, i.e. `group` to `owo$group`.

**Comment by BasiqueEvangelist** [comment]
I don't see any point in namespacing interface injected methods, especially the ones that have owo-specific types in their descriptors

File: src/main/java/io/wispforest/owo/mixin/itemgroup/ItemSettingsMixin.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+import org.spongepowered.asm.mixin.Mixin;
+
+import java.util.function.BiConsumer;
+
+@Mixin(Item.Settings.class)
+public class ItemSettingsMixin implements OwoItemSettingsExtension {
+    private OwoItemGroup owo$group = null;
+    private int owo$tab = 0;
+    // TODO: OwoItemGroup.DEFAULT_STACK_GENERATOR was inlined to not cause clinit issues. fix.
+    private BiConsumer<Item, ItemGroup.Entries> owo$stackGenerator = (item, stacks) -> stacks.add(item.getDefaultStack());
```
Comment (gliscowo) [inline src/main/java/io/wispforest/owo/mixin/itemgroup/ItemSettingsMixin.java]: 🤨

**Review by gliscowo — CHANGES_REQUESTED** [review]

File: src/main/java/io/wispforest/owo/mixin/itemgroup/ItemSettingsMixin.java (hunk 1 above)
Comment (BasiqueEvangelist) [reply src/main/java/io/wispforest/owo/mixin/itemgroup/ItemSettingsMixin.java]: ficksed

**Pull request merged**